src/shared/utilities/vsCodeUtils.ts

```diff
@@ -53,8 +55,37 @@ export function isExtensionActive(extId: string): boolean {
     return !!extension && extension.isActive
 }
 
-export function isExtensionInstalled(extId: string): boolean {
-    return !!vscode.extensions.getExtension(extId)
+/**
+ * Checks if an extension is installed and meets the version requirement
+ * @param minVersion The minimum semver required for the extension
+ */
+export function isValidExtensionInstalled(extId: string, minVersion?: string): boolean {
+    const ext = vscode.extensions.getExtension(extId)
+    if (ext === undefined) {
+        return false
+    }
+
+    return isExtensionValid(ext, minVersion)
+}
+
+export function isExtensionValid(ext: vscode.Extension<any>, minVersion?: string): boolean {
```

Comment:
i see that this is used in tests, else it would be unnecessary indirection. Though, we're not really testing much, except that semver does it it says it does :) So, a single test to "exercise" `isExtensionInstalled` would be enough imo, rather than deeper coverage. That could eliminate the need for this indirection.